### PR TITLE
Add mutex logging

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
Should be obvious what it does by looking at the newly added files.

### CODE REVIEW

**Comment by calmh** [comment]
Neat! At first I was opposed to the new* method, but the return in of a regular lock in the us debugged case more than compensates for that. There seems to be some gofmtery missing, and the copyright ear on the new files is old, otherwise :+1: :100:

File: internal/sync/sync.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+import (
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"sync"
+	"time"
+)
+
+type Mutex interface {
```
Comment (calmh) [inline internal/sync/sync.go]: We could use the sync.Locker and sync.RWLocker interfaces here. Should be exactly the same thing, possibly a teensy bit more documentative? I don't think there is one for the wait group though.

Comment (calmh) [inline internal/sync/sync.go]: Sorry, there is no RWLocker interface. Probably better to keep it as is above.

**Comment by AudriusButkevicius** [comment]
Should also write some tests.

**Comment by calmh** [comment]
(Could possibly copy whatever tests sync.Mutex etc already have, and check that they work under debugging. Seems slightly extracurricular though.)

**Comment by AudriusButkevicius** [comment]
I am more interesting it writing tests that they log stuff correctly.
We could also have a `STLOCKPANICTHRESHOLD` to perhaps include in integration tests to expose long-held locks.

**Comment by calmh** [comment]
Sounds reasonable

**Comment by calmh** [comment]
Basically that would obsolete the current deadlock detector

**Comment by AudriusButkevicius** [comment]
Sort of I guess, as some locks might be held deliberately for long periods of time.
We could do some funky getCaller() stuff get the function name where the lock is created and assign it to the lock (or <name>N where N is count of the duplicate name), then excluding locks by name from STLOCKPANICTHRESHOLD by name.

The reason I added this in the first place, is because if you make some API calls which access the model immediately after the GUI starts up, it seems to take multiple seconds to recover.

Haven't yet drilled down what's causing it, but I think it might be the summary service (which might be acceptable).

**Pull request merged**